### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by nmcclay**

### CODE REVIEW

**Comment by nmcclay** [comment]
I'm still tinkering with this @laurawhalin but I wanted to put up my progress so far.

File: _posts/2017-3-31-ember-component-showcase.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+category: development
+tags: development, javascript
+author_github: nmcclay
+date: 2017-03-27 06:00:00
+comments: true
+banner_image: ember-component-showcase-cover.jpg
+---
+
+## A note about documentation
+An often missed detail of successful documentation is maintaining the documentation to stay current with your implementation.  To this end developers have invented a great many inline documentation tools and utilities to allow them to update documentation in the same places that they change their code.
```
Comment (laurawhalin) [inline _posts/2017-3-31-ember-component-showcase.md]: outside the component code itself (word order mixup)

File: _posts/2017-3-31-ember-component-showcase.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    {{my-component value="foo"}}
+</div>
+    
+<pre class="source">
+     '{{my-component value="foo"}}'
+</pre>
+{% endraw %}
+{% endhighlight %}
+ 
+The reason for this is because ember's template compiler doesn't leave any scraps of pre-compiled HBS source laying around for reuse.
```
Comment (laurawhalin) [inline _posts/2017-3-31-ember-component-showcase.md]: Stay consistent with capitalization of Ember and HBS throughout file.

File: _posts/2017-3-31-ember-component-showcase.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+{% highlight handlebars %}
+{% raw %}
+{{#component-showcase}}
+    {{my-component value="foo"}}
+{{/component-showcase}}
+{% endraw %}
+{% endhighlight %}
+
+Inspired by [ember-freestyle's live style guide addon](https://github.com/chrislopresto/ember-freestyle), but implemented as a radically simpler and less opinionated documentation component.  We use this component internally at Wombat to document our own style guide component's, and now we are open sourcing this utility to the community.
```
Comment (laurawhalin) [inline _posts/2017-3-31-ember-component-showcase.md]: don't use possessive form for "components", or for "samples" in the next paragraph

File: _posts/2017-3-31-ember-component-showcase.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  {{s.source (array (hash title="route.js" language="JavaScript" src="var foo = 'foo'")) hideHTML=false}}
+  <p>Or after them!</p>
+{{/component-showcase}}
+{% endraw %}
+{% endhighlight %}
+
+## Next steps
+
+We still have a few things that need to be resolved in this addon before it's ready for prime time.
+As is usually the case with new addon's we need to beef up our testing to make sure we are covering more integration scenarios.  This is a little trickier than it would usually be when testing a component because of the nature of our component relying on ember's template compiler.
```
Comment (laurawhalin) [inline _posts/2017-3-31-ember-component-showcase.md]: don't use possessive form for "addons"

**Comment by nmcclay** [comment]
Pushed some updates, the repo is live now as well so everything looks good to go form my perspective!

**Pull request closed without merging**